src/ui/dialogs/omemoDevices.ts

```diff
@@ -39,15 +40,26 @@ class OmemoDeviceDialog {
 
          dom.find('.jsxc-omemo-peerdevices, .jsxc-omemo-owndevices').empty();
 
-         this.insertDevices(peerDevices, identityManager, dom.find('.jsxc-omemo-peerdevices'));
-         this.insertDevices(ownDevices, identityManager, dom.find('.jsxc-omemo-owndevices'));
-
+         this.insertDevices(contact.getJid().bare, peerDevices, identityManager, dom.find('.jsxc-omemo-peerdevices'));
+         this.insertDevices(contact.getAccount().getJID().bare, ownDevices, identityManager, dom.find('.jsxc-omemo-owndevices'));
          if (ownDevices.length > 1) {
             this.addCleanUpAction();
          }
+
       });
    }
 
+   private buildQRCode(jid: string, deviceid: string, key: string)
+   {
+        let qr = new QRCode();
+        qr.setTypeNumber(5);
+        qr.setErrorCorrectLevel(ErrorCorrectLevel.L);
+        qr.addData('xmpp:'+jid+'?omemo-sid-'+deviceid+'='+key.split(' ').join(''));
```

Comment:
Looking at the [commonly used url structure](https://github.com/psi-im/plugins/issues/59) (i will ask the omemo colleagues if there is any official documentation) you can verify all devices for one peer at once. This way the list will be a lot clearer.

Reply:
ok i changed it... :) combined the keys into one qr...